### ISSUE TITLE
torch_dreams.dreamer does not support gradient masks 

### ISSUE THREAD

**Issue by Mayukhdeb** [Mayukhdeb/torch-dreams#25]

**Comment by Mayukhdeb** [comment]
closing because we don't really need it for now

**Comment by eps696** [comment]
that's pity to hear; such extra functionality would distinguish this repo from the already existing well-developed ones (e.g. Lucent)

**Comment by Mayukhdeb** [comment]
Hi @eps696, the core differences between torch-dreams and lucent would be the intuitiveness of the interface and the amount of freedom the user gets (support for pretty much any objective function written by the user, using multiple models simultaneously, etc)

https://github.com/Mayukhdeb/torch-dreams/blob/6d05591d4681c7be06da0d23e13a836458c7cad4/torch_dreams/dreamer.py#L90

I really wish gradient masks were possible, but after running `loss.backward()`, when I check `img.grad`, it's still `None`. However, the original image parameter in the frequency domain does get it's gradients. 

In case you really want something like gradient masks, I've [tried out something similar](https://colab.research.google.com/github/Mayukhdeb/torch-dreams-notebooks/blob/main/docs_notebooks/hello_torch_dreams.ipynb#scrollTo=dCAY1-0hEMm8) by multiplying the image in the spatial domain by a mask to manipulate the outputs, but there you'll notice how the image gets white "residues" outside the mask (probably because it's optimized in the frequency domain)

I plan to come back to this issue when I get a better understanding of how gradients flow back through `torch.fft.irfft2()`

**Comment by eps696** [comment]
regarding freedom, that's partially true - lucid/lucent also allowed objectives combinations. it was bit more tricky than in torch-dreams indeed, but i guess the one who is ready to use custom functions in torch-dreams, is able to tweak lucent stuff too.
thanks for your thoughts anyway! looking forward very much to hearing of new/deeper tricks with fft manipulation as well, since i use this presentation method for another (somewhat similar) task [here](https://github.com/eps696/aphantasia).
